tree/dataframe/src/RNTupleDS.cxx

```diff
@@ -283,12 +284,14 @@ void RNTupleDS::AddField(const RNTupleDescriptor &desc, std::string_view colName
          fProtoFields.emplace_back(std::move(cardinalityField));
 
          for (const auto &f : desc.GetFieldIterable(fieldDesc.GetId())) {
-            AddField(desc, std::string(colName) + "." + f.GetFieldName(), f.GetId(), fieldInfos);
+            AddField(desc, std::string(colName) + "." + f.GetFieldName(), f.GetId(), fieldInfos, true);
```

Comment:
Unrelated to this change: I think with the revised untyped collections, this `if` branch can be merged with the `else` branch.